lib/sshkit/ssh/connection.ex

```diff
@@ -127,6 +100,7 @@ defmodule SSHKit.SSH.Connection do
     options =
       connection.options
       |> Keyword.put(:port, connection.port)
+      |> Keyword.put(:impl, connection.impl)
```

Comment:
Yup, absolutely true! When I am working on changes I tend to split it up into commits rather than PRs…

The changes to the implementation and tests at least had to go [hand-in-hand](https://youtu.be/aWbpT9hNAYQ), but I should've pulled out commits 2 & 3 into a separate PR to this one at least slightly smaller, even if not by much.

Thank you Tessi!